jetty-ee10/jetty-ee10-webapp/src/main/java/org/eclipse/jetty/ee10/webapp/WebAppContext.java

```diff
@@ -803,10 +803,12 @@ public Resource getWebInf() throws IOException
             return null;
 
         // Iw there a WEB-INF directory?
-        Resource webInf = getResourceBase().resolve("WEB-INF/");
+        Resource webInf = getResourceBase().resolve("WEB-INF/"); // TODO: what does this do in a collection?
```

Comment:
That's a very interesting TODO, and that contract needs to be clearly defined.

Reply:
There's no way to satisfy the Servlet spec with the existing Resource.resolve() contract and ResourceCollection.
It's super easy to wind up with things that are not part of the main archive.
This would be a Servlet specific contract issue, not a Resource specific contract issue (as we cannot push Servlet behaviors down to jetty-core)

Reply:
This needs to be punted to a future issue/PR.
Out of scope for this PR.
Leaving the TODO in place is good though.